routes/semen/semen.ts

```diff
@@ -23,4 +23,19 @@ router.post("", async (req: Request, res: Response) => {
     res.json(new SemenResponseWithBull(semenFromDb));
 })
 
+router.delete("/delete", async (req: Request, res: Response) => {
+    try{
+        const semenNumber: string = req.body.semenNumber;
+        await prisma.semen.delete({
+            where: {
+                number: semenNumber
+            }
+        })
+        res.json("Semen deleted")
+    } catch(error){
+        res.json("Semen does not exist")
```

Comment:
błąd w errorach - zamiast try catch powinnaś po prostu sprawdzić ifem czy dany objekt istnieje w bazie a jak nie to wyrzucić error (throw new ...). Używając try catch łapiesz wszystkie errory a nie tylko te że nie isnieje dany objekt